packages/ui-menu/src/Menu/MenuItemGroup/props.ts

```diff
@@ -38,19 +38,46 @@ import type {
   OtherHTMLAttributes
 } from '@instructure/shared-types'
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
+import type { MenuItemProps } from '../MenuItem/props'
+import React from 'react'
 
 type MenuGroupOwnProps = {
   label: React.ReactNode
   allowMultiple?: boolean
+  /**
+   * children of type `Menu.Item`, `Menu.Separator`
+   */
   children?: React.ReactNode // TODO: oneOf([MenuItem, MenuItemSeparator])
-  selected?: any // TODO: controllable(PropTypes.array, 'onSelect', 'defaultSelected')
-  defaultSelected?: any[]
-  onSelect?: (...args: any[]) => any
-  onMouseOver?: (...args: any[]) => any
-  onKeyDown?: (...args: any[]) => any
+  /**
+   * an array of the values (or indices by default) for the selected items
+   */
+  selected?: (string | number)[] // TODO: controllable(PropTypes.array, 'onSelect', 'defaultSelected')
+  /**
+   * an array of the values (or indices by default) for the selected items on initial render
+   */
+  defaultSelected?: (string | number)[]
+  /**
+   * call this function when a menu item is selected
+   */
+  onSelect?: (
+    e: React.MouseEvent,
+    updated: MenuItemProps['value'][],
+    selected: MenuItemProps['selected'],
+    item: MenuItem
+  ) => void
+  onMouseOver?: MenuItemProps['onMouseOver']
```

Comment:
I can see that `onKeyDown` was removed because it was unused, but it is an API change if we simply remove it, so I'd recommend deprecating it.

Reply:
Also, for now I think we should copy the types in cases like this, because the doc parser cannot resolve imported types and in this case it renders: `onMouseOver: intersection['onMouseOver']` in the Properties block. That sounds a bit lame, so I'd just copy  `(e: React.MouseEvent, args: MenuItem) => void` for now.

Reply:
I already did this with onBlur somewhere else. I don't think we can consider this an API change really